src/Admin/OptionsMenu.php

```diff
@@ -0,0 +1,490 @@
+<?php
+/**
+ * Class OptionsMenu
+ *
+ * @package Ampproject\Ampwp
+ */
+
+namespace AmpProject\AmpWP\Admin;
+
+use AMP_Analytics_Options_Submenu;
+use AMP_Core_Theme_Sanitizer;
+use AMP_Options_Manager;
+use AMP_Post_Type_Support;
+use AMP_Theme_Support;
+use AmpProject\AmpWP\Infrastructure\Conditional;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use AmpProject\AmpWP\Option;
+
+/**
+ * OptionsMenu class.
+ */
+class OptionsMenu implements Conditional, Service, Registerable {
+	/**
+	 * Handle for JS file.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @var string
+	 */
+	const ASSET_HANDLE = 'amp-settings';
+
+	/**
+	 * The AMP svg menu icon.
+	 *
+	 * @var string
+	 */
+	const ICON_BASE64_SVG = 'data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjIiIGhlaWdodD0iNjIiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+PHBhdGggZD0iTTQxLjYyODg2NjcgMjguMTYxNDMzM2wtMTMuMDA0NSAyMS42NDIxMzM0aC0yLjM1NmwyLjMyOTEzMzMtMTQuMTAxOS03LjIxMzcuMDA5M3MtLjA2ODIuMDAyMDY2Ni0uMTAwMjMzMy4wMDIwNjY2Yy0uNjQ5OTY2NyAwLTEuMTc1OTMzNC0uNTI1OTY2Ni0xLjE3NTkzMzQtMS4xNzU5MzMzIDAtLjI3OS4yNTkzNjY3LS43NTEyMzMzLjI1OTM2NjctLjc1MTIzMzNsMTIuOTYyMTMzMy0yMS42MTYzTDM1LjcyNDQgMTIuMTc5OWwtMi4zODgwMzMzIDE0LjEyMzYgNy4yNTA5LS4wMDkzcy4wNzc1LS4wMDEwMzMzLjExNDctLjAwMTAzMzNjLjY0OTk2NjYgMCAxLjE3NTkzMzMuNTI1OTY2NiAxLjE3NTkzMzMgMS4xNzU5MzMzIDAgLjI2MzUtLjEwMzMzMzMuNDk0OTY2Ny0uMjUwMDY2Ny42OTEzbC4wMDEwMzM0LjAwMTAzMzN6TTMxIDBDMTMuODc4NyAwIDAgMTMuODc5NzMzMyAwIDMxYzAgMTcuMTIxMyAxMy44Nzg3IDMxIDMxIDMxIDE3LjEyMDI2NjcgMCAzMS0xMy44Nzg3IDMxLTMxQzYyIDEzLjg3OTczMzMgNDguMTIwMjY2NyAwIDMxIDB6IiBmaWxsPSIjYTBhNWFhIiBmaWxsLXJ1bGU9ImV2ZW5vZGQiLz48L3N2Zz4=';
+
+	/**
+	 * GoogleFonts instance.
+	 *
+	 * @var GoogleFonts
+	 */
+	private $google_fonts;
+
+	/**
+	 * ReaderThemes instance.
+	 *
+	 * @var ReaderThemes
+	 */
+	private $reader_themes;
+
+	/**
+	 * Check whether the conditional object is currently needed.
+	 *
+	 * @return bool Whether the conditional object is needed.
+	 */
+	public static function is_needed() {
+		if ( ! is_admin() ) {
+			return false;
+		}
+
+		/**
+		 * Filter whether to enable the AMP settings.
+		 *
+		 * @since 0.5
+		 * @param bool $enable Whether to enable the AMP settings. Default true.
+		 */
+		$short_circuit = apply_filters( 'amp_options_menu_is_enabled', true );
```

Comment:
Just realized that we can now re-use this `is_needed` method to determine whether we register a top-level "AMP DevTools" menu in the `amp_validated_url` post type and the `amp_validation_error` taxonomy, as opposed to currently we only use the `manage_options` capability check. In the rare scenario where someone has the menu disabled but does have dev tools enabled, they'd not be able to access the validation errors screens. Something to take note of for the future, but not necessarily needing to be done now.